File: python_files/ex12e6.py
class Motor:
    def __init__(self, cilindradas, potencia):
        self.cilindradas = cilindradas
        self.potencia = potencia

    def __str__(self):
        return f"[{self.cilindradas}cc, {self.potencia} HP]"

class Carro:
    def __init__(self,marca,modelo,ano,cor,motor):
        self.marca=marca
        self.modelo=modelo
        self.ano=ano
        self.cor=cor
        self.velocidade=0
        self.motor=motor
    
    def acelerar(self, incremento):
        self.velocidade += incremento
    
    def frear(self, decremento):
        self.velocidade-=decremento
        if self.velocidade < 0:
            self.velocidade = 0

    def exibirVelocidade(self):
        print(f"{self.modelo} está a {self.velocidade} km/h.")


    def __str__(self):
        return f"marca={self.marca},modelo={self.modelo}, ano={self.ano}, cor={self.cor}, motor={self.motor}"


motor = Motor(2000, 150)
carro1=Carro('Fiat','Uno',2010,'cinza',motor)   
carro2=Carro('Ford','Fiesta',2014,'preto',motor)
carro3=Carro('Chevrolet','Onix',2020,'prata',motor)
print(carro1)
print(carro2)
print(carro3)

carro3.acelerar(100)
carro3.exibirVelocidade()
carro3.frear(10)
carro3.exibirVelocidade()

File: go_files/ex12e6.go
package main
import ("fmt")

type Carro struct {
  Marca string
  Modelo string
  Ano int
  Cor string
  Velocidade int
}

func (c *Carro) acelerar(incremento int) {
  c.Velocidade += incremento
}

func (c *Carro) frear(incremento int) {
  c.Velocidade -= incremento
  if c.Velocidade < 0 {
    c.Velocidade=0
  }
}

func (c *Carro) exibirVelocidade() {
  fmt.Printf("%s está a %d km/h.\n", c.Modelo, c.Velocidade)
}

func main() {

  carro1 := Carro{
		Marca:  "Fiat",
		Modelo: "Uno",
		Ano:    2010,
		Cor:    "Cinza",
	}

	carro2 := Carro{
		Marca:  "Ford",
		Modelo: "Fiesta",
		Ano:    2014,
		Cor:    "Preto",
	}

	carro3 := Carro{
		Marca:  "Chevrolet",
		Modelo: "Onix",
		Ano:    2020,
		Cor:    "Prata",
	}

  informacoesCarro(carro1,1)

  informacoesCarro(carro2,2)

  informacoesCarro(carro3,3)

  carro1.exibirVelocidade()
  carro1.acelerar(20)
  carro1.exibirVelocidade()
  carro1.frear(15)
  carro1.exibirVelocidade()
  carro1.frear(30)
  carro1.exibirVelocidade()


}

func informacoesCarro(carro Carro,numCarro int) {
	fmt.Print("carro",numCarro,":\n")
	fmt.Println("marca:", carro.Marca)
	fmt.Println("modelo:", carro.Modelo)
	fmt.Println("ano:", carro.Ano)
	fmt.Println("cor:", carro.Cor)
}